### PR TITLE
Action Manager | Action Enabled State Handling

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Introduces new API functions to handle the enabled state for an Action in the Action Manager:

- InstallEnabledStateCallback enables to set a callback whose return value will determine whether the action is enabled or disabled;
- IsActionEnabled allows to query the enabled state of an action;
- UpdateAction is now expanded to update both the enabled and checked state.

This PR also adds some unit tests to verify the functionality, and slightly alters existing ones to match the changes.

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp [hunk 1]
```diff
@@ -242,6 +242,19 @@ namespace AzToolsFramework
         return AZ::Success();
     }
 
+    ActionManagerBooleanResult ActionManager::IsActionEnabled(const AZStd::string& actionIdentifier)
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp:245]: question: Could this be marked `const`? (even if this is an EBus interface it's now generally good practice to add const correctness where possible as we may use this through an AZ::Interface)

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManagerInterface.h [hunk 2]
```diff
@@ -60,14 +66,26 @@ namespace AzToolsFramework
...
+        //! @param handler The handler function that should be called when the action is triggered.
+        //! @return A successful outcome object, or a string with a message detailing the error in case of failure.
         virtual ActionManagerOperationResult RegisterAction(
             const AZStd::string& contextIdentifier,
             const AZStd::string& actionIdentifier,
             const ActionProperties& properties,
             AZStd::function<void()> handler
         ) = 0;
-
+
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManagerInterface.h:80]: uber nit: Can we remove the trailing whitespace please 🙂

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManagerInterface.h [hunk 3]
```diff
@@ -77,37 +95,81 @@ namespace AzToolsFramework
...
         virtual ActionManagerOperationResult TriggerAction(const AZStd::string& actionIdentifier) = 0;
 
         //! Retrieve a QAction via its identifier.
+        //! @param actionIdentifier The identifier for the action to retrieve.
+        //! @return A raw pointer to the QAction, or nullptr if the action could not be found.
         virtual QAction* GetAction(const AZStd::string& actionIdentifier) = 0;
+        
+        //! Retrieve a QAction via its identifier (const version).
+        //! @param actionIdentifier The identifier for the action to retrieve.
+        //! @return A raw const pointer to the QAction, or nullptr if the action could not be found.
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManagerInterface.h:158]: Nice updates with the API comments! 🙂

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorAction.h [hunk 4]
```diff
@@ -43,7 +43,13 @@ namespace AzToolsFramework
         //! Returns the pointer to the action.
         QAction* GetAction();
 
-        //! Calls the callback to update the action's checked state, if any.
+        //! Sets the enabled state callback for the action.
+        void SetEnabledStateCallback(AZStd::function<bool()> enabledStateCallback);
+
+        //! Returns true if the EditorAction has an enabled state callback set, false otherwise.
+        bool HasEnabledStateCallback();
```
Comment (hultonha) [inline Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorAction.h:50]: update: This should probably be `const` (also should we also have an `IsEnabled` call too? Or is that handled elsewhere?)

**Review by hultonha — APPROVED** [review]
Looks great to me! I only have minor comments/nits 👍

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp (hunk 1 above)
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/ActionManager.cpp:245]: Makes sense, fixed!

File: Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorAction.h (hunk 4 above)
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/ActionManager/Action/EditorAction.h:50]: const added, IsEnabled added, changed the ActionManager IsActionEnabled to use IsEnabled.

**Review by cgalvan — APPROVED** [review]
Approved (+1 to Tom's suggestions on const)

**Pull request merged**